src/libraries/System.Formats.Tar/tests/TarEntry/TarEntry.ExtractToFile.Tests.cs

```diff
@@ -92,5 +94,68 @@ public void ExtractToFile_Link_Throws(TarEntryFormat format, TarEntryType entryT
 
             Assert.Equal(0, Directory.GetFileSystemEntries(root.Path).Count());
         }
+
+        public static IEnumerable<object[]> GetFormatsAndFiles()
+        {
+            foreach (TarEntryFormat format in new[] { TarEntryFormat.Ustar, TarEntryFormat.Pax, TarEntryFormat.Gnu })
+            {
+                foreach (TarEntryType entryType in new[] { TarEntryType.RegularFile, TarEntryType.Directory })
+                {
+                    yield return new object[] { format, entryType };
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(GetFormatsAndFiles))]
+        public void Extract(TarEntryFormat format, TarEntryType entryType)
+        {
+            using TempDirectory root = new TempDirectory();
+
+            (string entryName, string destination, PosixTarEntry entry) = Prepare_Extract(root, format, entryType);
+
+            entry.ExtractToFile(destination, overwrite: true);
+
+            Verify_Extract(destination, entry, entryType);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetFormatsAndFiles))]
+        public async Task Extract_Async(TarEntryFormat format, TarEntryType entryType)
+        {
+            using TempDirectory root = new TempDirectory();
+
+            (string entryName, string destination, PosixTarEntry entry) = Prepare_Extract(root, format, entryType);
+
+            await entry.ExtractToFileAsync(destination, overwrite: true);
+
+            Verify_Extract(destination, entry, entryType);
+        }
+
+        private (string, string, PosixTarEntry) Prepare_Extract(TempDirectory root, TarEntryFormat format, TarEntryType entryType)
+        {
+            string entryName = entryType.ToString();
+            string destination = Path.Join(root.Path, entryName);
+
+            PosixTarEntry entry = InvokeTarEntryCreationConstructor(format, entryType, entryName) as PosixTarEntry;
```

Comment:
If you include V7 in the member data above, you can avoid casting to PosixTarEntry.